### DISCUSSION TITLE
https://mokkapps.de/vue-tips/mutate-your-state-inside-of-the-provider-using-provide-and-inject?source=weeklyVueNews&campaign=145

### DISCUSSION THREAD

**Discussion by giscus** [Mokkapps/mokkapps-de-discussions#10] [Announcements]
# https://mokkapps.de/vue-tips/mutate-your-state-inside-of-the-provider-using-provide-and-inject?source=weeklyVueNews&campaign=145

When using reactive provide/inject values, it is recommended to keep any mutations to reactive state inside of the provider whenever possible.

https://mokkapps.de/vue-tips/mutate-your-state-inside-of-the-provider-using-provide-and-inject?source=weeklyVueNews&campaign=145

**Comment by dixhuit** [comment]
You have a typo in your final Child.vue example (it's in your email newsletter too). Should `<button @click="increment">Increment</button>` not be `<button @click="incrementCount">Increment</button>`? (see handler function name)

Great tip regardless though! :)

**Reply by Mokkapps** [reply]
Thanks, will fix that asap 👍🏻💚
